From 05f447dad791c3a4e8e04fb0116d0b2f04716d66 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Sun, 6 Jan 2013 13:07:11 +0200
Subject: [PATCH 389/609] Added IPC support to GP board

	The IPC device registration was missing from the GP
	board machine start function.

	Bug fix: Please merge to stable branch

Change-Id: Ic0cfe0e3e94e1699267c37cde14947da0e432782
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Seif Mazareeb <seif@marvell.com>
---
 arch/arm/mach-armadaxp/core.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-armadaxp/core.c b/arch/arm/mach-armadaxp/core.c
index 9285d10..4ecc300 100755
--- a/arch/arm/mach-armadaxp/core.c
+++ b/arch/arm/mach-armadaxp/core.c
@@ -550,18 +550,19 @@ static struct platform_device axp_i2c1 = {
 };
 #endif
 
-
 void __init armadaxp_i2c0_init(void)
 {
 	if (mvUnitMapIsMine(I2C0) == MV_TRUE)
 		platform_device_register(&axp_i2c0);
 }
 
+#ifdef CONFIG_FB_DOVE
 void __init armadaxp_i2c1_init(void)
 {
 	if (mvUnitMapIsMine(I2C1) == MV_TRUE)
 		platform_device_register(&axp_i2c1);
 }
+#endif
 
 /**********
  * UART-0 *
@@ -1873,6 +1874,10 @@ static void __init axp_gp_init(void)
 	eth_init();
 #endif
 
+#ifdef CONFIG_MV_IPC_NET
+	platform_device_register(&mv_ipc_net);
+#endif
+
 #ifdef CONFIG_FB_DOVE
 	kw_lcd0_dmi.dram = &armadaxp_mbus_dram_info;
 	if (lcd_panel) {
-- 
1.7.9.5

